packages/block-library/src/comment-template/edit.js

```diff
@@ -23,11 +28,89 @@ const TEMPLATE = [
 	[ 'core/comment-edit-link' ],
 ];
 
-function CommentTemplateInnerBlocks() {
+function CommentTemplateInnerBlocks( {
+	comments,
+	activeBlockContext,
+	setActiveBlockContext,
+	blocks,
+} ) {
 	const innerBlocksProps = useInnerBlocksProps( {}, { template: TEMPLATE } );
-	return <li { ...innerBlocksProps } />;
+	return (
+		<li { ...innerBlocksProps }>
+			{ innerBlocksProps.children }
+			{ comments.length > 0 ? (
+				<RenderComments
+					blockContexts={ comments }
+					activeBlockContext={ activeBlockContext }
+					setActiveBlockContext={ setActiveBlockContext }
+					blocks={ blocks }
+				/>
+			) : null }
+		</li>
+	);
 }
 
+/**
+ * Component that renders a list of (nested) comments. It is called recursively
+ * in its own body.
+ *
+ * @param {Object} props                         Component props.
+ * @param {Array}  [props.blockProps]            - Props from parent's `useBlockProps()`.
+ * @param {Array}  [props.blockContexts]         - Array of comment objects.
+ * @param {Array}  [props.activeBlockContext]    - The block that is currently active.
+ * @param {Array}  [props.setActiveBlockContext] - The setter for activeBlockContext.
+ * @param {Array}  [props.blocks]                - Array of blocks returned from
+ *                                               getBlocks() in parent .
+ * @return {WPElement}                 		List of comments.
+ */
+const RenderComments = ( {
```

Comment:
Nitpick: how about we rename it? 
```suggestion
const CommentsList = ( {
```

Somehow, the verb in the component name raises a flag in my head :she